# origin: Debian (herbert)
# cset: n/a
# inclusion: not submitted
# description: remove extra definition of cmpxchg in drmP.h
# revision date: 2004-09-05

diff -urN kernel-source-2.4.26-1/drivers/char/drm-4.0/drmP.h kernel-source-2.4.26-2/drivers/char/drm-4.0/drmP.h
--- kernel-source-2.4.26-1/drivers/char/drm-4.0/drmP.h	2004-02-21 14:47:05.000000000 +1100
+++ kernel-source-2.4.26-2/drivers/char/drm-4.0/drmP.h	2004-04-24 16:11:31.000000000 +1000
@@ -220,39 +220,7 @@
      (__typeof__(*(ptr))) __cmpxchg((ptr), (unsigned long)_o_,		 \
 				    (unsigned long)_n_, sizeof(*(ptr))); \
   })
-
-#elif __i386__
-static inline unsigned long __cmpxchg(volatile void *ptr, unsigned long old,
-				      unsigned long new, int size)
-{
-	unsigned long prev;
-	switch (size) {
-	case 1:
-		__asm__ __volatile__(LOCK_PREFIX "cmpxchgb %b1,%2"
-				     : "=a"(prev)
-				     : "q"(new), "m"(*__xg(ptr)), "0"(old)
-				     : "memory");
-		return prev;
-	case 2:
-		__asm__ __volatile__(LOCK_PREFIX "cmpxchgw %w1,%2"
-				     : "=a"(prev)
-				     : "q"(new), "m"(*__xg(ptr)), "0"(old)
-				     : "memory");
-		return prev;
-	case 4:
-		__asm__ __volatile__(LOCK_PREFIX "cmpxchgl %1,%2"
-				     : "=a"(prev)
-				     : "q"(new), "m"(*__xg(ptr)), "0"(old)
-				     : "memory");
-		return prev;
-	}
-	return old;
-}
-
-#define cmpxchg(ptr,o,n)						\
-  ((__typeof__(*(ptr)))__cmpxchg((ptr),(unsigned long)(o),		\
-				 (unsigned long)(n),sizeof(*(ptr))))
-#endif /* i386 & alpha */
+#endif /* alpha */
 #endif
 
 				/* Macros to make printk easier */
